### DISCUSSION TITLE
[SYCL] How to link a pre-compiled device image?

### DISCUSSION THREAD

**Discussion by shixiyue** [intel/llvm#5708] [General]
I read the documentation about [Dynamic linking of device code](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/SharedLibraries.md#dynamic-linking-of-device-code), which describe a scenario of using functions defined in a pre-compiled device image provided by the user. 

May I know what are the compiler commands for this scenario? The documentation says that 
> user adds device image with definition of LibDeviceFunc to the fat object via special compiler option (like -fsycl-add-targets 

But I don't understand how to do that in practice. Could you provide an example for the compilation command? Also, what if the pre-compiled device image is for an architecture different from the host's? Does the dynamic linking of device code still work in the case? Thank you!

**Comment by romanovvlad** [comment]
@shixiyue 
As far as I know it is not supported yet. @AlexeySachkov Could you please comment?

**Reply by shixiyue** [reply]
Got it. Thank you!

**Reply by shixiyue** [reply]
@romanovvlad @AlexeySachkov Would there be any method to link a device library that is in native device binary? I got an error when I tried to link a library compiled for the device's architecture which is different from the host:
Relocations in generic ELF (EM: 243)
libDevice.a: error adding symbols: File in wrong format
